### PR TITLE
Move cardano-node config/launch setup to separate library

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

#1458 

# Overview

- [x] I Moved the `cardano-node-integration:Cardano.Wallet.Byron.Config` module to `cardano-wallet-byron-launch:Cardano.Wallet.Byron.Launch`. I changed `Config` -> `Launch` for consistency with [`Jormungandr`](https://github.com/input-output-hk/cardano-wallet/blob/5aa90d1f445a481ab95a06deb577b10ae9bc65c2/lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/Launch.hs#L11).

# Comments

Such that we can use this code from the restoration benchmark when we add it.

### CODE REVIEW

File: lib/byron/cardano-wallet-byron.cabal [hunk 1]
```diff
@@ -71,6 +71,59 @@ library
...
+    , async
+    , bytestring
+    , cardano-crypto-wrapper
+    , cardano-ledger
+    , cardano-wallet-core
+    , cardano-wallet-byron
+    , cardano-wallet-launcher
+    , cborg
+    , containers
+    , contra-tracer
```
Comment (Anviking) [inline lib/byron/cardano-wallet-byron.cabal]: Weeder will definitely fail.

File: lib/byron/test/launch/Cardano/Wallet/Byron/Launch.hs [hunk 2]
```diff
@@ -1,9 +1,8 @@
 {-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE NamedFieldPuns #-}
-{-# LANGUAGE TemplateHaskell #-}
 {-# LANGUAGE TypeApplications #-}
 
-module Cardano.Wallet.Byron.Config
```
Comment (Anviking) [inline lib/byron/test/launch/Cardano/Wallet/Byron/Launch.hs]: Should add a module doc comment similar to the jormungandr one.

**Comment by KtorZ** [comment]
Why though :no_mouth:  ?

**Comment by Anviking** [comment]
@KtorZ Oops, I only wrote it in the commit: To be able to re-use it from benchmarks.

**Comment by KtorZ** [comment]
Well, I prefer it only written in the commit than only on the PR's body :smile: 
Ok. Didn't strike me as obvious at first :+1:

**Review by KtorZ — APPROVED** [review]

**Comment by Anviking** [comment]
Could be that this is not possible. Not sure yet.
```
Cabal-simple_mPHDZzAJ_2.4.0.1_ghc-8.6.5: internal error when calculating
--
  | transitive package dependencies.
```
I've run in to this in the past (regarding haddock), but I thought it was under different circumstances.

**Comment by KtorZ** [comment]
We can't have multiple libraries in a single cabal file. The feature is actually broken. I'll suggest to move the module into the main Byron library.

**Review by KtorZ — APPROVED** [review]

**Review by rvl — APPROVED** [review]
Looks fine, especially if tests/benchmarks still pass.

But you might want to add a note that this is intended to be used for testing, not as an official way of starting cardano-wallet-byron -- Unless we make an effort to keep this implementation in sync with the javascript cardano-launcher.

**Comment by Anviking** [comment]
```
Failures:
      src/Test/Integration/Framework/DSL.hs:1103:9:
      1) API Specifications, STAKE_POOLS_JOIN/QUIT_02 - Passphrase must be text, Join
           expected: Status {statusCode = 204, statusMessage = "No Content"}
            but got: Status {statusCode = 500, statusMessage = "Internal Server Error"}
           from the following response: Left (HttpException ResponseTimeout)
      To rerun use: --match "/API Specifications/STAKE_POOLS_JOIN/QUIT_02 - Passphrase must be text/Join/"
    Randomized with seed 1483408781
```

should be unrelated
bors r+

**Pull request merged**